framework/transport/ieee1905_transport/ieee1905_transport_packet_processing.cpp

```diff
@@ -536,6 +579,64 @@ bool Ieee1905Transport::forward_packet(Packet &packet)
     return true;
 }
 
+/**
+ * @brief forward a 1905.1 packet.
+ * 
+ * Basically a wrapper to forward_packet_single, which also takes into account the reliable
+ * multicast procedure - according to section 15.1 in the Multi-AP specification, a reliable multicast
+ * transmission includes sending the packet with the relay bit set to a multicast address once,
+ * and once for each discovered Multi-AP agent in the network as unicast
+ * with the same MID and the relay bit unset.
+ * 
+ * @param packet packet to forward
+ * @return true on success
+ * @return false on failure (one or more packets failed in transmission)
+ */
+bool Ieee1905Transport::forward_packet(Packet &packet)
+{
+    // First, forward the packet as is.
+    bool success = forward_packet_single(packet);
+    if (!success) {
+        MAPF_DBG("Failed to forward packet " << packet);
+        return false;
+    }
+
+    // Check if this packet should also go through reliable multicast:
+    // 1. Destination is multicast
+    // 2. Message type is topology notification, which is currently the only
+    //    message which should be sent using reliable multicast. See Multi-AP
+    //    specification Table 5
+    auto should_send_reliable_multicast = [&](const Packet &packet) -> bool {
```

Comment:
why create a lamda when you use it only once?

Reply:
For better readability